File: tests/rspec/pages/design/template_variables.rb
class TemplateVariables < ControlPanelPage
  set_url_matcher /design\/variables/

  element :create_new_button, 'fieldset.right a.btn.action'

  element :keyword_search, '.filters ul li:first-child input'
  element :perpage_filter, '.filters ul li:nth-child(2) a'

  sections :variables, 'table.app-listing tbody tr' do
    element :name, 'td:first-child'
    element :all_sites, 'td:nth-child(2)'
    section :manage, 'td:nth-child(3)' do
      element :edit, '.toolbar .edit a'
      element :find, '.toolbar .find a'
    end
    element :bulk_action_checkbox, 'td:nth-child(4) input'
  end

  def load
    self.open_dev_menu
    click_link 'Templates'
    click_link 'Template Variables'
  end

end

File: tests/cypress/cypress/elements/pages/design/TemplateVariables.js
import ControlPanel from '../ControlPanel'

class TemplateVariables extends ControlPanel {
    constructor() {
        super()
        this.url = 'admin.php?/cp/design/variables';

        this.elements({
            "create_new_button": '.title-bar__extra-tools a.button--action',

            "keyword_search": '.title-bar .filter-bar .search-input__input',
            "perpage_filter": '.title-bar .filters ul li:nth-child(2) a',

            "variables": '.table-responsive table tbody tr',
            // "name": 'td:first-child',
            // "all_sites": 'td:nth-child(2)',
            // "manage": 'td:nth-child(3)',
            //   "edit": '.toolbar .edit a',
            //   "find": '.toolbar .find a',
            // "bulk_action_checkbox": 'td:nth-child(4) input',
        })
    }
}
export default TemplateVariables;